src/lib.rs

```diff
@@ -972,123 +973,42 @@ pub fn max_level() -> LevelFilter {
     unsafe { mem::transmute(MAX_LOG_LEVEL_FILTER.load(Ordering::Relaxed)) }
 }
 
-/// Sets the global logger.
+/// Sets the global logger to a `Box<Log>`.
 ///
-/// The `make_logger` closure is passed a `MaxLevel` object, which the
-/// logger should use to keep the global maximum log level in sync with the
-/// highest log level that the logger will not ignore.
+/// This is a simple convenience wrapper over `set_logger`, which takes a
+/// `Box<Log>` rather than a `&'static Log`. See the documentation for
+/// [`set_logger`] for more details.
 ///
-/// This function may only be called once in the lifetime of a program. Any log
-/// events that occur before the call to `try_set_logger` completes will be
-/// ignored.
-///
-/// This function does not typically need to be called manually. Logger
-/// implementations should provide an initialization method that calls
-/// `try_set_logger` internally.
-///
-/// Requires the `use_std` feature (enabled by default).
+/// Requires the `use_std` feature.
 ///
 /// # Errors
 ///
-/// This function fails to set the global logger if it has already
-/// been called before.
-///
-/// # Example
-///
-/// Implements a custom logger `ConsoleLogger` which prints to stdout.
-/// In order to use the logging macros, `ConsoleLogger` implements
-/// the [`Log`] trait and has to be installed via `set_logger`.
+/// An error is returned if a logger has already been set.
 ///
-/// ```rust
-/// # #[macro_use]
-/// # extern crate log;
-/// #
-/// use log::{Record, Level, Metadata, LevelFilter, SetLoggerError};
-///
-/// struct ConsoleLogger;
-///
-/// impl log::Log for ConsoleLogger {
-///     fn enabled(&self, metadata: &Metadata) -> bool {
-///         metadata.level() <= Level::Info
-///     }
-///
-///     fn log(&self, record: &Record) {
-///         if self.enabled(record.metadata()) {
-///             println!("Rust says: {} - {}", record.level(), record.args());
-///         }
-///     }
-///     fn flush(&self) {}
-/// }
-///
-/// fn try_init() -> Result<(), SetLoggerError> {
-///     log::try_set_logger(|max_log_level| {
```

Comment:
Both this example and the one below were ignoring the result of try_set_logger....